src/main/java/com/google/cloud/teleport/v2/bigquery-gcs/src/main/java/com/google/cloud/teleport/v2/bigquerygcs/BigQueryToParquet.java

```diff
@@ -0,0 +1,261 @@
+ * IMAGE_NAME=my-image-name
+ * TARGET_GCR_IMAGE=gcr.io/${PROJECT}/${IMAGE_NAME}
+ * BASE_CONTAINER_IMAGE=my-base-container-image
+ * BASE_CONTAINER_IMAGE_VERSION=my-base-container-image-version
+ * APP_ROOT=/path/to/app-root
+ * COMMAND_SPEC=/path/to/command-spec
+ *
+ * # Build and upload image
+ * mvn clean package \
+ * -Dimage=${TARGET_GCR_IMAGE} \
+ * -Dbase-container-image=${BASE_CONTAINER_IMAGE} \
+ * -Dbase-container-image.version=${BASE_CONTAINER_IMAGE_VERSION} \
+ * -Dapp-root=${APP_ROOT} \
+ * -Dcommand-spec=${COMMAND_SPEC}
+ *
+ * # Create an image spec in GCS that contains the path to the image
+ * {
+ *    "docker_template_spec": {
+ *       "docker_image": $TARGET_GCR_IMAGE
+ *     }
+ *  }
+ *
+ * # Execute template:
+ * API_ROOT_URL="https://dataflow.googleapis.com"
+ * TEMPLATES_LAUNCH_API="${API_ROOT_URL}/v1b3/projects/${PROJECT}/templates:launch"
+ * JOB_NAME="bigquery-to-parquet-`date +%Y%m%d-%H%M%S-%N`"
+ *
+ * time curl -X POST -H "Content-Type: application/json"     \
+ *     -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+ *     "${TEMPLATES_LAUNCH_API}"`
+ *     `"?validateOnly=false"`
+ *     `"&dynamicTemplate.gcsPath=${BUCKET_NAME}/path/to/image-spec"`
+ *     `"&dynamicTemplate.stagingLocation=${BUCKET_NAME}/staging" \
+ *     -d '
+ *      {
+ *       "jobName":"'$JOB_NAME'",
+ *       "parameters": {
+ *           "tableRef":"'$TABLE'",
+ *           "bucket":"'$BUCKET_NAME/results'",
+ *           "numShards":"5",
+ *           "fields":"field1,field2"
+ *        }
+ *       }
+ *      '
+ * </pre>
+ */
+public class BigQueryToParquet {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BigQueryToParquet.class);
+
+    /** File suffix for file to be written. */
+    private static final String fileSuffix = ".parquet";
+
+    /** Factory to create BigQueryStorageClients. */
+    public static class BigQueryStorageClientFactory {
+      BigQueryStorageClient create() {
+          try {
+          return BigQueryStorageClient.create();
+        } catch (IOException e) {
+          LOG.error("Error connecting to BigQueryStorage API: " + e.getMessage());
+          throw new RuntimeException(e);
+        }
+      }
+  }
+
+    /** Factory to create ReadSessions. */
+    static class ReadSessionFactory {
+
+        /**
+         * Creates ReadSession for schema extraction.
+         * @param tableString String that represents table to export from.
+         * @param tableReadOptions TableReadOptions that specify any fields in the table to filter on.
+         * @return session ReadSession object that contains the schema for the export.
+         */
+        ReadSession create(
+                BigQueryStorageClient client,
+                String tableString,
+                TableReadOptions tableReadOptions) {
+            TableReference tableReference = BigQueryHelpers.parseTableSpec(tableString);
+            String parent = "projects/" + tableReference.getProjectId();
```

Comment:
May want to call this parentProjectId or something to make it less ambiguous.

Reply:
Done.